devito/parameters.py

```diff
@@ -136,3 +157,54 @@ def init_configuration():
             configuration.update(k, keys)
 
     configuration.initialize()
+
+
+def add_sub_configuration(sub_configuration, sub_env_vars_mapper=None):
+    init_configuration(sub_configuration, sub_env_vars_mapper or {})
+    setattr(configuration, sub_configuration.name, sub_configuration)
+
+
+def print_defaults():
+    """Print the environment variables accepted by Devito, their default value,
+    as well as all of the accepted values."""
+    from devito.logger import info
+    for k, v in env_vars_mapper.items():
+        info('%s: %s. Default: %s' % (k, configuration._accepted[v],
+                                      configuration._defaults[v]))
+
+
+def print_state():
+    """Print the current configuration state."""
+    from devito.logger import info
+    for k, v in configuration.items():
+        info('%s: %s' % (k, v))
+
+
+def infer_cpu():
```

Comment:
Probably obvious from name, but docstring would still be nice.

Reply:
Done